composer/trainer/ddp.py

```diff
@@ -414,60 +245,10 @@ def ddp_sync_context(self, state: State, is_final_microbatch: bool):
                                     p.grad = p.grad / state.world_size
 
         else:
-            raise ValueError("Unknown sync strategy", self.ddp_sync_strategy)
-
-
-@dataclass
-class StoreHparams(hp.Hparams, ABC):
-
-    @abstractmethod
-    def initialize_object(self, is_main: bool, world_size: int) -> torch.distributed.Store:
-        pass
-
-
-@dataclass
-class TCPStoreHparams(StoreHparams):
-    host_name: str = hp.optional(doc="Rank 0 address", default="127.0.0.1")
-    port: int = hp.optional(doc="Rank 0 port", default=43297)
-
-    def initialize_object(self, is_main: bool, world_size: int) -> torch.distributed.Store:
-        return torch.distributed.TCPStore(self.host_name, self.port, world_size, is_main)
-
-
-@dataclass
-class FileStoreHparams(StoreHparams):
-    file_name: str = hp.required(doc="Path to store file")
-
-    def initialize_object(self, is_main: bool, world_size: int) -> torch.distributed.Store:
-        return torch.distributed.FileStore(self.file_name, world_size)
+            raise ValueError("Unknown sync strategy", self.sync_strategy)
 
 
 @dataclass
 class DDPHparams(hp.Hparams):
-    hparams_registry = {
-        "store": {
-            "tcp": TCPStoreHparams,
-            "file": FileStoreHparams,
-        }
-    }
-
-    store: StoreHparams = hp.optional(doc="Store", default_factory=TCPStoreHparams)
-    node_rank: int = hp.optional(doc="Node ID for multi-node training", default=0)
-    num_nodes: int = hp.optional(doc="Number of nodes used for training", default=1)
-    fork_rank_0: bool = hp.optional(
-        doc="Whether to fork the local rank 0 process, or use the existing process for rank 0 training.",
-        default=False,
-    )
+    sync_strategy: Optional[str] = hp.optional(doc="Strategy for DDP syncing", default=None)
```

Comment:
Would love to have the default be something reasonable or explicit somewhere

Reply:
I'll make the comment a bit more descriptive, but the issue is that the default depends on whether `find_unused_parameters` is on.